includes/admin/class-wc-rest-stripe-account-keys-controller.php

```diff
@@ -0,0 +1,208 @@
+		if ( ! empty( $result ) && ! preg_match( $validate_options['regex'], $param ) ) {
+			return new WP_Error( 400, $validate_options['error_message'] );
+		}
+		return true;
+	}
+
+	public function validate_publishable_key( $param, $request, $key ) {
+		return $this->validate_stripe_param(
+			$param,
+			$request,
+			$key,
+			[
+				'regex'         => '/^pk_live_/',
+				'error_message' => __( 'The "Live Publishable Key" should start with "pk_live", enter the correct key.', 'woocommerce-gateway-stripe' ),
+			]
+		);
+	}
+
+	public function validate_secret_key( $param, $request, $key ) {
+		return $this->validate_stripe_param(
+			$param,
+			$request,
+			$key,
+			[
+				'regex'         => '/^[rs]k_live_/',
+				'error_message' => __( 'The "Live Secret Key" should start with "sk_live" or "rk_live", enter the correct key.', 'woocommerce-gateway-stripe' ),
+			]
+		);
+	}
+
+	public function validate_test_publishable_key( $param, $request, $key ) {
+		return $this->validate_stripe_param(
+			$param,
+			$request,
+			$key,
+			[
+				'regex'         => '/^pk_test_/',
+				'error_message' => __( 'The "Test Publishable Key" should start with "pk_test", enter the correct key.', 'woocommerce-gateway-stripe' ),
+			]
+		);
+	}
+
+	public function validate_test_secret_key( $param, $request, $key ) {
+		return $this->validate_stripe_param(
+			$param,
+			$request,
+			$key,
+			[
+				'regex'         => '/^[rs]k_test_/',
+				'error_message' => __( 'The "Test Secret Key" should start with "sk_test" or "rk_test", enter the correct key.', 'woocommerce-gateway-stripe' ),
+			]
+		);
+	}
+
+	/**
+	 * Update the data.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function set_account_keys( WP_REST_Request $request ) {
+		$publishable_key      = $request->get_param( 'publishable_key' );
+		$secret_key           = $request->get_param( 'secret_key' );
+		$webhook_secret       = $request->get_param( 'webhook_secret' );
+		$test_publishable_key = $request->get_param( 'test_publishable_key' );
+		$test_secret_key      = $request->get_param( 'test_secret_key' );
+		$test_webhook_secret  = $request->get_param( 'test_webhook_secret' );
+
+		$settings                         = $this->get_stripe_settings();
+		$settings['publishable_key']      = is_null( $publishable_key ) ? $settings['publishable_key'] : $publishable_key;
+		$settings['secret_key']           = is_null( $secret_key ) ? $settings['secret_key'] : $secret_key;
+		$settings['webhook_secret']       = is_null( $webhook_secret ) ? $settings['webhook_secret'] : $webhook_secret;
+		$settings['test_publishable_key'] = is_null( $test_publishable_key ) ? $settings['test_publishable_key'] : $test_publishable_key;
+		$settings['test_secret_key']      = is_null( $test_secret_key ) ? $settings['test_secret_key'] : $test_secret_key;
+		$settings['test_webhook_secret']  = is_null( $test_webhook_secret ) ? $settings['test_webhook_secret'] : $test_webhook_secret;
```

Comment:
Should we allow `null` keys? Not inputting anything in a field would probably equal to `""`, not `null`.

Maybe we should require all keys to be present depending on the mode used: live or test, and have a separate endpoint for clearing keys? :shrug:

Reply:
if input is "", then save it as "". // serve as erase
if input is null, then keep the original
if input is some text, update it. 

> Maybe we should require all keys to be present depending on the mode used
Yea, i am not sure. We can probably adjust this when we work on the UI.

Reply:
Ah, gotcha! Tested and this is indeed how it works. Thanks! :100: